lib/galaxy/datatypes/binary.py

```diff
@@ -462,6 +473,58 @@ def to_archive(self, trans, dataset, name=""):
         file_paths.append(dataset.metadata.bam_index.file_name)
         return zip(file_paths, rel_paths)
 
+    def get_chunk( self, trans, dataset, offset=0, ck_size=None ):
+        index_file = dataset.metadata.bam_index
+        with pysam.AlignmentFile( dataset.file_name, "rb", index_filename=index_file.file_name ) as bamfile:
+            ck_size = 300  # 300 lines
+            ck_data = ""
+            last_read = 0
+            header_line_count = 0
+            if offset == 0:
+                ck_data = bamfile.text
```

Comment:
I think for now we should cut them off as you're doing now. In the future we could work with the `comment_lines` MetadataElement and be a little smarter about this, i.e let the user choose to see the full header. I don't think this is too important for the start.